sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/jsonExpressions.scala

```diff
@@ -502,111 +496,119 @@ case class JsonTuple(children: Seq[Expression])
     }
   }
 
+  @transient
+  private lazy val evaluator: JsonTupleEvaluator = JsonTupleEvaluator(fieldExpressions.length)
+
   override def eval(input: InternalRow): IterableOnce[InternalRow] = {
```

Comment:
is it possible to move more code into `JsonTupleEvaluator`?

Reply:
like passing the entire `InternalRow` to `JsonTupleEvaluator#eval`?

Reply:
Okay, let me give it a try.